src/shared_realm.cpp

```diff
@@ -575,17 +575,23 @@ void Realm::notify()
     m_is_sending_notifications = true;
     auto cleanup = util::make_scope_exit([this]() noexcept { m_is_sending_notifications = false; });
 
+    // The SharedRealm could be closed in the callbacks. Keep a ref locally to check if Realm gets closed.
+    auto this_ref = shared_from_this();
     if (m_shared_group->has_changed()) { // Throws
         if (m_binding_context) {
             m_binding_context->changes_available();
+            if (this_ref->is_closed()) return;
```

Comment:
Calling these via `this_ref->` suggests that it's somehow different from `this`, when it's actually fine to just call `is_closed()` directly.